src/NuGet.Core/NuGet.Protocol/Plugins/PluginFactory.cs

```diff
@@ -455,5 +479,45 @@ private static void WriteCommonLogMessages(IPluginLogger logger)
             logger.Write(new ProcessLogMessage(logger.Now));
             logger.Write(new ThreadPoolLogMessage(logger.Now));
         }
+
+        public async Task<IPlugin> GetOrCreateNetToolsPluginAsync(string filePath, IEnumerable<string> arguments, IRequestHandlers requestHandlers, ConnectionOptions options, CancellationToken sessionCancellationToken)
```

Comment:
We discussed this briefly. 

The name here talks about a net tools plugin, but it's really about an executable file, which doesn't really need to a dotnet tool, it can just be a .bat file.

We can make the names match that.